src/persist-client/src/internal/state.rs

```diff
@@ -833,20 +898,28 @@ where
 
     pub fn clone_apply<R, E, WorkFn>(
         &self,
-        build_version: &Version,
+        cfg: &PersistConfig,
         work_fn: &mut WorkFn,
     ) -> ControlFlow<E, (R, Self)>
     where
-        WorkFn: FnMut(SeqNo, &mut StateCollections<T>) -> ControlFlow<E, R>,
+        WorkFn: FnMut(SeqNo, &PersistConfig, &mut StateCollections<T>) -> ControlFlow<E, R>,
     {
         let mut new_state = State {
-            applier_version: build_version.clone(),
+            applier_version: cfg.build_version.clone(),
             shard_id: self.shard_id,
             seqno: self.seqno.next(),
+            walltime_ms: (cfg.now)(),
+            hostname: cfg.hostname.clone(),
             collections: self.collections.clone(),
             _phantom: PhantomData,
         };
-        let work_ret = work_fn(new_state.seqno, &mut new_state.collections)?;
+        // Make sure walltime_ms is strictly increasing, in case clocks are
```

Comment:
why does it need to be strictly increasing? seems like it could be useful to capture when different processes have clock drift

Reply:
because of this bit I buried in the commit message

> Also, persist state is a definite collection. This makes it a _reclocked_ definite collection.

interestingly, we could also use this combined with heartbeat timestamps to maybe figure out some amount of clock offset

I snuck this third commit in here because I figured it would be uncontroversial. if that's not true, then I think i'd rather pull it out and discuss async rather than trying to rush the conversation so we can get the first two commits into this week's release

Reply:
thought about this more and I'm just going to pull this third commit into a separate PR

Reply:
👍 separate PR sounds good